### ISSUE TITLE
Minecraft 1.16 support

### ISSUE THREAD

**Issue by questforboobs** [mircokroon/minecraft-world-downloader#23]
Unsurprinsingly, the latest release doesn't work with Minecraft 1.16 prerelease servers. Here is an error log.

```
Successfully authenticated user with Mojang session server.
User identity confirmed with Mojang.
Enabled encryption
Enabled compression
java.lang.ArrayIndexOutOfBoundsException: Index 28 out of bounds for length 28
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider.readString(DataTypeProvider.java:94)
        at packets.builder.ClientBoundLoginPacketBuilder.lambda$new$2(ClientBoundLoginPacketBuilder.java:33)
        at packets.builder.PacketBuilder.build(PacketBuilder.java:46)
        at packets.DataReader.readPackets(DataReader.java:161)
        at packets.DataReader.pushData(DataReader.java:116)
        at proxy.ProxyServer.lambda$run$7(ProxyServer.java:113)
        at proxy.ProxyServer.attempt(ProxyServer.java:138)
        at proxy.ProxyServer.lambda$run$9(ProxyServer.java:110)
        at proxy.ProxyServer.attempt(ProxyServer.java:138)
        at proxy.ProxyServer.run(ProxyServer.java:68)
java.lang.ArrayIndexOutOfBoundsException: Index 52 out of bounds for length 52
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider.readString(DataTypeProvider.java:94)
        at packets.builder.ClientBoundLoginPacketBuilder.lambda$new$2(ClientBoundLoginPacketBuilder.java:34)
        at packets.builder.PacketBuilder.build(PacketBuilder.java:46)
        at packets.DataReader.readPackets(DataReader.java:161)
        at packets.DataReader.pushData(DataReader.java:116)
        at proxy.ProxyServer.lambda$run$7(ProxyServer.java:113)
        at proxy.ProxyServer.attempt(ProxyServer.java:138)
        at proxy.ProxyServer.lambda$run$9(ProxyServer.java:110)
        at proxy.ProxyServer.attempt(ProxyServer.java:138)
        at proxy.ProxyServer.run(ProxyServer.java:68)
Login success: [redacted] logged in with uuid [redacted]
Using protocol of game version 1.15.0 (725)
java.lang.ArrayIndexOutOfBoundsException: Index 18 out of bounds for length 18
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider$1.read(DataTypeProvider.java:166)
        at java.base/java.io.InputStream.read(Unknown Source)
        at java.base/java.io.DataInputStream.readFully(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at se.llbit.nbt.StringTag.read(StringTag.java:46)
        at se.llbit.nbt.NamedTag.read(NamedTag.java:58)
        at packets.DataTypeProvider.readNbtTag(DataTypeProvider.java:163)
        at game.data.chunk.version.Chunk_1_14.parseHeightMaps(Chunk_1_14.java:39)
        at game.data.chunk.Chunk.parse(Chunk.java:257)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
java.lang.RuntimeException: VarInt lacks bytes! We may be out of sync now.
        at packets.DataReader.readVarInt(DataReader.java:68)
        at packets.DataTypeProvider.readVarInt(DataTypeProvider.java:100)
        at game.data.chunk.Chunk.parse(Chunk.java:263)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
Chunk could not be parsed!
java.lang.ArrayIndexOutOfBoundsException: Index 18 out of bounds for length 18
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider$1.read(DataTypeProvider.java:166)
        at java.base/java.io.InputStream.read(Unknown Source)
        at java.base/java.io.DataInputStream.readFully(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at se.llbit.nbt.StringTag.read(StringTag.java:46)
        at se.llbit.nbt.NamedTag.read(NamedTag.java:58)
        at packets.DataTypeProvider.readNbtTag(DataTypeProvider.java:163)
        at game.data.chunk.version.Chunk_1_14.parseHeightMaps(Chunk_1_14.java:39)
        at game.data.chunk.Chunk.parse(Chunk.java:257)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
java.lang.RuntimeException: VarInt lacks bytes! We may be out of sync now.
        at packets.DataReader.readVarInt(DataReader.java:68)
        at packets.DataTypeProvider.readVarInt(DataTypeProvider.java:100)
        at game.data.chunk.Chunk.parse(Chunk.java:263)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
Chunk could not be parsed!
java.lang.ArrayIndexOutOfBoundsException: arraycopy: last source index 9 out of bounds for byte[2]
        at java.base/java.lang.System.arraycopy(Native Method)
        at packets.DataTypeProvider.readByteArray(DataTypeProvider.java:79)
        at packets.DataTypeProvider.readDouble(DataTypeProvider.java:184)
        at packets.builder.ServerBoundGamePacketBuilder.lambda$new$0(ServerBoundGamePacketBuilder.java:13)
        at packets.builder.PacketBuilder.build(PacketBuilder.java:46)
        at packets.DataReader.readPackets(DataReader.java:161)
        at packets.DataReader.pushData(DataReader.java:116)
        at proxy.ProxyServer.lambda$run$4(ProxyServer.java:93)
        at proxy.ProxyServer.attempt(ProxyServer.java:138)
        at proxy.ProxyServer.lambda$run$6(ProxyServer.java:90)
        at java.base/java.lang.Thread.run(Unknown Source)
java.lang.ArrayIndexOutOfBoundsException: Index 18 out of bounds for length 18
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider$1.read(DataTypeProvider.java:166)
        at java.base/java.io.InputStream.read(Unknown Source)
        at java.base/java.io.DataInputStream.readFully(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at se.llbit.nbt.StringTag.read(StringTag.java:46)
        at se.llbit.nbt.NamedTag.read(NamedTag.java:58)
        at packets.DataTypeProvider.readNbtTag(DataTypeProvider.java:163)
        at game.data.chunk.version.Chunk_1_14.parseHeightMaps(Chunk_1_14.java:39)
        at game.data.chunk.Chunk.parse(Chunk.java:257)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
java.lang.RuntimeException: VarInt lacks bytes! We may be out of sync now.
        at packets.DataReader.readVarInt(DataReader.java:68)
        at packets.DataTypeProvider.readVarInt(DataTypeProvider.java:100)
        at game.data.chunk.Chunk.parse(Chunk.java:263)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
Chunk could not be parsed!
java.lang.ArrayIndexOutOfBoundsException: Index 18 out of bounds for length 18
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider$1.read(DataTypeProvider.java:166)
        at java.base/java.io.InputStream.read(Unknown Source)
        at java.base/java.io.DataInputStream.readFully(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at se.llbit.nbt.StringTag.read(StringTag.java:46)
        at se.llbit.nbt.NamedTag.read(NamedTag.java:58)
        at packets.DataTypeProvider.readNbtTag(DataTypeProvider.java:163)
        at game.data.chunk.version.Chunk_1_14.parseHeightMaps(Chunk_1_14.java:39)
        at game.data.chunk.Chunk.parse(Chunk.java:257)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
java.lang.RuntimeException: VarInt lacks bytes! We may be out of sync now.
        at packets.DataReader.readVarInt(DataReader.java:68)
        at packets.DataTypeProvider.readVarInt(DataTypeProvider.java:100)
        at game.data.chunk.Chunk.parse(Chunk.java:263)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
Chunk could not be parsed!
java.lang.ArrayIndexOutOfBoundsException: Index 18 out of bounds for length 18
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider$1.read(DataTypeProvider.java:166)
        at java.base/java.io.InputStream.read(Unknown Source)
        at java.base/java.io.DataInputStream.readFully(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at se.llbit.nbt.StringTag.read(StringTag.java:46)
        at se.llbit.nbt.NamedTag.read(NamedTag.java:58)
        at packets.DataTypeProvider.readNbtTag(DataTypeProvider.java:163)
        at game.data.chunk.version.Chunk_1_14.parseHeightMaps(Chunk_1_14.java:39)
        at game.data.chunk.Chunk.parse(Chunk.java:257)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
java.lang.RuntimeException: VarInt lacks bytes! We may be out of sync now.
        at packets.DataReader.readVarInt(DataReader.java:68)
        at packets.DataTypeProvider.readVarInt(DataTypeProvider.java:100)
        at game.data.chunk.Chunk.parse(Chunk.java:263)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
Chunk could not be parsed!
java.lang.ArrayIndexOutOfBoundsException: Index 18 out of bounds for length 18
        at packets.DataTypeProvider.readNext(DataTypeProvider.java:65)
        at packets.DataTypeProvider$1.read(DataTypeProvider.java:166)
        at java.base/java.io.InputStream.read(Unknown Source)
        at java.base/java.io.DataInputStream.readFully(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at java.base/java.io.DataInputStream.readUTF(Unknown Source)
        at se.llbit.nbt.StringTag.read(StringTag.java:46)
        at se.llbit.nbt.NamedTag.read(NamedTag.java:58)
        at packets.DataTypeProvider.readNbtTag(DataTypeProvider.java:163)
        at game.data.chunk.version.Chunk_1_14.parseHeightMaps(Chunk_1_14.java:39)
        at game.data.chunk.Chunk.parse(Chunk.java:257)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
java.lang.RuntimeException: VarInt lacks bytes! We may be out of sync now.
        at packets.DataReader.readVarInt(DataReader.java:68)
        at packets.DataTypeProvider.readVarInt(DataTypeProvider.java:100)
        at game.data.chunk.Chunk.parse(Chunk.java:263)
        at game.data.chunk.ChunkFactory.readChunkDataPacket(ChunkFactory.java:178)
        at game.data.chunk.ChunkFactory.run(ChunkFactory.java:128)
Chunk could not be parsed!
```

**Comment by mircokroon** [comment]
Yup, probably won't fix it until the full release it out though

**Comment by NurMarvin** [comment]
Well, Minecraft 1.16 released yesterday, so it would be nice to have 1.16 support :)

**Comment by mircokroon** [comment]
Now works in the latest [release](https://github.com/mircokroon/minecraft-world-downloader/releases)!
